pyap_beauhurst/parser.py

```diff
@@ -66,69 +58,69 @@ def parse(self, text):
         return results
 
     def _parse_address(self, match):
-        '''Parses address into parts'''
+        """Parses address into parts"""
         if isinstance(match, str):
             # If the address is passed as a match it saves foing the match twice
             match = utils.match(self.rules, match, flags=re.VERBOSE | re.U)
         if match:
             match_as_dict = match.groupdict()
-            match_as_dict.update({'country_id': self.country})
+            match_as_dict.update({"country_id": self.country})
             # combine results
             cleaned_dict = self._combine_results(match_as_dict)
-            cleaned_dict['match_start'] = match.start()
-            cleaned_dict['match_end'] = match.end()
+            cleaned_dict["match_start"] = match.start()
+            cleaned_dict["match_end"] = match.end()
             # create object containing results
             return address.Address(**cleaned_dict)
 
         return False
 
     @staticmethod
     def _combine_results(match_as_dict):
```

Comment:
Realise this isn't your code but seems like a crazy way of doing things

Reply:
which part in particular is crazy?

Reply:
Some things, maybe not for  this initial PR:
- I think I'd prefer `_parse_address` to return None rather than False if there are no matches.
- I think the match start and end indices should probably be thrown away (maybe they could be kept if you ask them to be kept, but for what we are using it for, we only care about the address on its own, not really in the context of the larger text)
-I think most of the weirdness around combining results here is from accommodating both French and English address structures in the Canadian regex. This is not immediately applicable to what we are doing, but I don't really want to cut Canada out to neaten up a few lines.

Reply:
I think I'll look at how best to accommodate the multiple postcode regex group names in a future PR